include/vcpkg/base/util.h

```diff
@@ -106,6 +112,56 @@ namespace vcpkg::Util
         }
     }
 
+    template<class Map, class Key>
+    typename Map::mapped_type copy_or_default(const Map& map, Key&& key)
+    {
+        const auto it = map.find(static_cast<Key&&>(key));
+        if (it == map.end())
+        {
+            return typename Map::mapped_type{};
+        }
+
+        return it->second;
+    }
+
+    template<class Map, class Key>
+    const typename Map::mapped_type& value_or_default(const Map& map,
```

Comment:
Add an `= delete` for the `&&` overload to prevent accidental "ref to temporary"